### PR TITLE
feature S3C-4213: Add docs for using vault

### PR DESCRIPTION
**Pull request by miniscruff**
Reference vault in the docs as the env var is not mentioned.

### CODE REVIEW

File: docs/GETTING_STARTED.rst [hunk 1]
```diff
@@ -91,6 +91,16 @@ This starts a Zenko CloudServer on port 8000.
 
 The default access key is accessKey1. The secret key is verySecretKey1.
 
+Run it with Vault user management
```
Comment (wabernat) [inline docs/GETTING_STARTED.rst]: Run CloudServer with Vault User Management

Comment (miniscruff) [reply docs/GETTING_STARTED.rst]:
```suggestion
Run CloudServer with Vault User Management
```

**Comment by rachedbenmustapha** [comment]
With Vault not being open source, I don't know how relevant it is to this particular part of the doc. I know I get annoyed when some open source project references something I can't have access to, if only to know what it is.

**Comment by miniscruff** [comment]
> With Vault not being open source, I don't know how relevant it is to this particular part of the doc. I know I get annoyed when some open source project references something I can't have access to, if only to know what it is.

Hmm, I was thinking that at first too. I was thinking that since it is referenced in code it might as well be referenced in the readme. But I wasn't sure on this. Maybe instead I could add this to the vault readme.

File: README.md [hunk 2]
```diff
@@ -134,6 +134,7 @@ The default access key is accessKey1 with
 a secret key of verySecretKey1.
 
 ## Run it with Vault user management
+Note: Vault is a private product and you will need access to it separately.
```
Comment (wabernat) [inline README.md]:
```suggestion
Note: Vault is proprietary and must be accessed separately.
```

Comment (miniscruff) [reply README.md]: I spent like 5 minutes thinking about how to best describe this, before I realized I had no idea what terms to use and just went with whatever... thanks

Comment (wabernat) [reply README.md]: No worries, matey. Words are what I do!

**Review by wabernat — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Review by saitharunthodupunuru — APPROVED** [review]

**Pull request merged**